cats/cats-redis/src/test/groovy/com/netflix/spinnaker/cats/redis/cluster/ClusteredSortAgentSchedulerSpec.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 THL A29 Limited, a Tencent company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.cats.redis.cluster
+
+import com.netflix.spinnaker.cats.agent.AgentExecution
+import com.netflix.spinnaker.cats.agent.CachingAgent;
+import com.netflix.spinnaker.cats.agent.ExecutionInstrumentation
+import com.netflix.spinnaker.cats.cluster.DefaultAgentIntervalProvider
+import com.netflix.spinnaker.cats.cluster.NodeStatusProvider
+import com.netflix.spinnaker.cats.test.TestAgent
+import redis.clients.jedis.Jedis
+import redis.clients.jedis.JedisPool
+import spock.lang.Specification
+import spock.lang.Subject
+
+class ClusteredSortAgentSchedulerSpec extends Specification {
```

Comment:
It's perhaps too late for this PR, but we'd much prefer new code -- even test code -- in any language but groovy :)  Java or kotlin please.

Reply:
Ok, I will try.

Reply:
Done.

Reply:
Thank you!